nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/NiFiPropertiesLoader.java

```diff
@@ -211,6 +220,45 @@ public NiFiProperties get() {
         return instance;
     }
 
+    private void checkForDuplicates(File file) {
+        Map<String, String> properties = new HashMap<>();
+        Map<String, Set<String>> duplicateProperties = new HashMap<>();
+        Pattern keyPattern = Pattern.compile("([^!#=][^=]*)=(.*)");
+
+        if (file == null) {
+            throw new IllegalArgumentException("NiFi properties file missing or unreadable");
+        }
+        // Scan the properties file for duplicate keys
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(file))) {
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                Matcher matcher = keyPattern.matcher(line);
+
+                if (matcher.matches()) {
+                    String key = matcher.group(1);
+                    String value = matcher.group(2);
+
+                    String existingValue = properties.put(key, value);
+
+                    if (existingValue != null && !existingValue.equals(value)) {
```

Comment:
Did you want to log a warning when there are duplicates but the values are the same? I believe this wasn't noted as an exceptional case and the startup should continue.

Reply:
I considered this 'no harm, no foul' as the application is not affected by duplicate keys with the same value. However, I agree that a warning level message might be a good idea to help users (who are paying attention) keep their nifi.properties file clean.
I added a warning log message in cases of duplicate keys with the same value.
Full disclosure: I also changed the wording of the error message when values are not the same.